src/inference/src/dev/core_impl.cpp

```diff
@@ -243,7 +243,9 @@ bool ov::is_config_applicable(const std::string& user_device_name, const std::st
     return false;
 }
 
-ov::Parsed ov::parseDeviceNameIntoConfig(const std::string& deviceName, const AnyMap& config) {
+ov::Parsed ov::parseDeviceNameIntoConfig(const std::string& deviceName,
+                                         const AnyMap& config,
+                                         const bool keep_batch_property) {
```

Comment:
Could we use common name for this kind of properties which is not part of HW plugin properties but needed by compile_model() for HW plugin? Batch plugin properties are current cases of this kind of properties.

Reply:
Do I understand correctly? It is required to differentiate AUTO BATCH properties only. For other properties like cache_dir, there is not requirement to clean up here.